applications/u2f/u2f_nfc.c

```diff
@@ -0,0 +1,232 @@
+#include "u2f_nfc.h"
+#include "furi_hal.h"
+
+#define TAG "U2F_NFC"
+
+// TODO: determine actual max payload length
+#define U2F_NFC_MAX_PAYLOAD_LEN (7680)
+
+#define U2F_NFC_MAX_PAYLOAD_PER_PACKET (256)
+
+typedef enum {
+    WorkerEvtReserved = (1 << 0),
+    WorkerEvtStop = (1 << 1),
+} WorkerEvtFlags;
+
+typedef enum {
+    StateUnselected,
+    StateSelected,
+    StateSending,
+} U2fNfcState;
+
+struct U2fNfc {
+    FuriThread* thread;
+    U2fData* u2f_instance;
+    U2fNfcState state;
+    uint8_t payload[U2F_NFC_MAX_PAYLOAD_LEN];
+    uint16_t payload_len;
+    uint16_t payload_cursor;
+};
+
+static bool u2f_nfc_callback(
+    uint8_t* buff_rx,
+    uint16_t buff_rx_len,
+    uint8_t* buff_tx,
+    uint16_t* buff_tx_len,
+    uint32_t* data_type,
+    void* context) {
+    U2fNfc* u2f_nfc = context;
+    *data_type = FURI_HAL_NFC_TXRX_DEFAULT;
+    buff_tx[0] = 0x02;
+    *buff_tx_len = 0;
+    FURI_LOG_D(TAG, "Something detected of length %d", buff_rx_len);
+    if(buff_rx_len >= 24) {
+        uint8_t pcb = buff_rx[0];
+        uint8_t cla = buff_rx[1];
+        uint8_t ins = buff_rx[2];
+        FURI_LOG_D(TAG, "pcb: 0x%02x cla: 0x%02x ins: 0x%02x", pcb, cla, ins);
+        if((pcb & ~0x03) != 0) {
+            buff_tx[1] = 0x68;
+            buff_tx[2] = 0x00;
+            *buff_tx_len = 24;
+            FURI_LOG_E(TAG, "pcb 0x%02x not supported", pcb);
+            return true;
+        }
+        if(cla != 0x00) {
+            buff_tx[1] = 0x68;
+            buff_tx[2] = 0x00;
+            *buff_tx_len = 24;
+            FURI_LOG_D(TAG, "cla 0x%02x not supported", cla);
+            return true;
+        }
+        if(u2f_nfc->state == StateUnselected && ins == 0xa4) {
+            if(memcmp(&buff_rx[3], "\x04\x00\x08\xa0\x00\x00\x06\x47\x2f\x00\x01", 11) != 0) {
```

Comment:
Can you please add more insights what the byte sequences mean? Datasheets, documentation or comments.